### ISSUE TITLE
YAML returns `null` or `undefined`

### ISSUE THREAD

**Issue by oscarotero** [denoland/deno_std#3426]
**Describe the bug**

This issue comes up here: https://github.com/lumeland/lume/pull/431#issuecomment-1578393330

Parsing an empty string returns `undefined`, but a string with only a comment, returns `null`.

There's a [relevant issue in the js-yaml library](https://github.com/nodeca/js-yaml/issues/567).

**Steps to Reproduce**

Run the following in REPL mode:

```
> import {parse} from "https://deno.land/std@0.189.0/yaml/parse.ts";
> parse('')
undefined
> parse('# test')
null
```

**Expected behavior**

I'd expect to return always `undefined` in all cases (or `null` if you prefer, but not both).

**Environment**

- OS: macOS
- deno version: 1.34.0
- std version: all

**Comment by not-my-profile** [comment]
It's not just about comments the same odd behavior can also be observed with just newlines:

```
> parse('\n')
undefined
> parse('\n\n')
null
```

**Comment by not-my-profile** [comment]
I guess this can be closed now that #3442 has been merged? (thanks btw!)
